forum/users/serializers.py

```diff
@@ -82,15 +84,37 @@ def validate_role(self, value):
             raise serializers.ValidationError("Invalid role.")
         return value
 
-    def validate_first_name(self, value):
-        if len(value) > 30:
-            raise serializers.ValidationError("First name must not exceed 30 characters.")
-        return value
 
-    def validate_last_name(self, value):
-        if len(value) > 30:
-            raise serializers.ValidationError("Last name must not exceed 30 characters.")
-        return value
+VALID_TOKEN_ROLES = [Role.STARTUP, Role.INVESTOR]
+
+
+class CustomTokenObtainPairSerializer(TokenObtainPairSerializer):
+    role = serializers.ChoiceField(choices=[(role.value, role.name.lower()) for role in VALID_TOKEN_ROLES])
```

Comment:
The role validation logic is written in the `validate` method. Moving this to the `ChoiceField` directly improves readability and reusability.
```
role = serializers.ChoiceField(choices=[(role.value, role.name.lower()) for role in VALID_TOKEN_ROLES], required=True)
```